codes/go/pkg/tree_node.go

```diff
@@ -9,6 +9,9 @@ import (
 	"fmt"
 )
 
+// nodeNotExist represents node don't exist.
+const nodeNotExist int = -1
```

Comment:
Does this definition work if we have a node with val = -1?

Reply:
I see. I have find a better way to solve this.